core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -635,8 +636,10 @@ public long getRawSize() {
       // backwards compatibility
       if (version.equals(API_VERSION_1)) {
         LOG.trace("Found a version 1 file to read.");
-        decryptionParams = new NoFileEncrypter().getDecryptionParameters();
         this.decrypter = new NoFileDecrypter();
+        try (FileEncrypter enc = new NoFileEncrypter()) {
+          decryptionParams = enc.getDecryptionParameters();
+        }
       } else {
         // read crypto parameters and get decrypter
         this.in.seek(offsetCryptoParameters);
```

Comment:
We might want to put some checks in to ensure that the CryptoService is open so we don't write null values to the file.